pyanaconda/core/kickstart/specification.py

```diff
@@ -102,8 +105,28 @@ def __init__(self, specification):
 
     def registerSectionData(self, name, data):
         """Register data used by a section."""
-        obj = data()
-        setattr(self, name, obj)
+        if isinstance(data, tuple):
+            # Multiple data objects (section instances) stored in a list
+            data, data_list_name = data
+
+            class SectionDataListString():
+                def __init__(self, data_list):
+                    self._data_list = data_list
+
+                def __str__(self):
+                    retval = []
+                    for data_obj in self._data_list:
+                        if isinstance(data_obj, data):
+                            retval.append(data_obj.__str__())
+                    return "".join(retval)
```

Comment:
Is there a reason to hide this class like this? Usually we are creating classes as top level.

Reply:
The reason is I don't feel the class is not ready / safe to be reused. It feels very single-purpose to me. I'll think more about it.

Reply:
Maybe also better name would be `SectionDataListStrWrapper`. And put it at top level in the file?